app/scripts/services/specialtygettingservice.js

```diff
@@ -55,6 +56,20 @@ angular.module('admissionSystemApp')
       return deferredId.promise;
     };
 
+    service.searchSpecialty = function (str) {
+      fillSpecialtiesArray().then(function () {
+        var specialty = null;
+        angular.forEach(specialties, function (item) {
+          if (parseInt(item.id) === parseInt(str)) {
+            specialty = item;
+            return false;
+          }
+        });
+        deferredSpec.resolve(specialty);
+      });
+      return deferredSpec.promise;
```

Comment:
We don't need other promise variable there.

```javascript
service.searchSpecialty = function (str) {
    return fillSpecialtiesArray().then(function () {
        ...
        return specialty
    }
}
```

and the same in other functions.